File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
		int max = 0;

		// 저점은 임시로 맨 첫 원소 값
		int min = prices[0];

		for (int i = 0; i < prices.length; i++) {
			min = Math.min(min, prices[i]);

			max = Math.max(max, prices[i] - min);
		}

		return max;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var max = 0
        var min = prices[0]

        for (num in prices) {
            min = min.coerceAtMost(num)

            max = max.coerceAtLeast(num - min)
        }
        return max
    }
}